server/platform/shared/filestore/s3store.go

```diff
@@ -652,6 +654,12 @@ func (b *S3FileBackend) listDirectory(path string, recursion bool) ([]string, er
 		if trimmed != "" {
 			paths = append(paths, trimmed)
 		}
+		count++
+	}
+	// Check if only one item was returned and it matches the path prefix
+	if count == 1 && len(paths) > 0 && strings.TrimRight(path, "/") == strings.TrimRight(paths[0], "/") {
```

Comment:
No need to do `strings.TrimRight` for `paths[0]` because it's already done inside the for loop.